auth/base/src/main/java/org/wildfly/security/auth/principal/NamePrincipal.java

```diff
@@ -93,4 +93,36 @@ public boolean equals(final NamePrincipal obj) {
     public String toString() {
         return name;
     }
+
+    /**
+     * Attempt to convert the given principal to a {@code NamePrincipal}.
+     *
+     * @param principal the original principal
+     * @return the {@code NamePrincipal} or {@code null} if the principal cannot be converted
+     */
+    public static NamePrincipal asNamePrincipal(Principal principal) {
+        if (principal instanceof NamePrincipal) {
+            return (NamePrincipal) principal;
+        }
+        return isNamePrincipal(principal) ? new NamePrincipal(principal.getName()) : null;
+    }
+
+    /**
+     * Check if the given principal could be converted to a {@code NamePrincipal}.
+     *
+     * @param principal the original principal
+     * @return {@code true} if the principal can be converted to a {@code NamePrincipal} and {@code false} otherwise
+     */
+    public static boolean isNamePrincipal(Principal principal) {
```

Comment:
The changes generally look good but I am not sure about the method name.  It is a bit more than just checking if it is a name principal.

Reply:
Good point, how about `convertibleToNamePrincipal` instead?

Reply:
+1, I've updated this to use `from` and `isConvertibleTo`. Thanks!